doc/thingpedia-nl-support.md

```diff
@@ -67,10 +154,80 @@ action share(in req status: String #_[prompt="What do you want to post?"])
 ```
 
 By default, the slot filling question will be “What's the value of status?”.
-Now with the prompt provided, Almond will ask “What do you want to post” instead. 
+Now with the prompt provided, Almond will ask “What do you want to post?” instead. 
+
+### Confirmation 
+Before a program is executed, Almond will confirm with the user 
+to make sure it understands the command correctly. 
+The confirmation sentence comes from the `confirmation` annotation of the function.
+Similar to the `utterances` annotation for example commands, 
+use noun phrases for queries and verb phrases in the imperative form for actions. 
+Examples can be found in all the tutorials. 
+
+Note that the confirmation sentence should include all the __required__ input parameter
+with `$param` or `${param}`.
 
 ### Output format
 When a stream or a query is combined with the action `notify`, 
 the returned results will be presented to the user in Almond. 
 By default, Almond will simply list the value of all the output parameters one by one 
-in the order they are declared in the function signature.
\ No newline at end of file
+in the order they are declared in the function signature.
+
+You can customize the output format with the `formatted` annotation.
+The annotation takes a list of Objects. 
+Each Object represents one response and each response will be presented to users in the same order in the list. 
+Four types of responses are supported: `text`, `picture`, `rdl`, and `code`.
+
+#### Text response 
+A text response returns a simple text message.
+It has only one property: `text`. 
+E.g., `{ type="text",text="Works in ${industry}" }` (in [Tutorial 3](/doc/thingpedia-tutorial-linkedin.md)).
```

Comment:
As a shorthand, you can specify a text message with a bare string. That is, the following are equivalent:
```
#_[formatted=["Works in ${industry}"]]
#_[formatted=[{ type="text",text="Works in ${industry}" }]
```

Inputs and output parameters of the function can be referred to using the placeholder syntax. The following options can be specified, using the syntax `${param:opt}`:
- `time` (for type `Date`): display only the time portion of the date
- `date` (for type `Date`): display only the date (day, month, year) portion of the date
- `iso-date` (for type `Date`): display the date in [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) format
- `%` (for type `Number`): multiply the number by 100 before displaying
- a measurement unit (for type `Measure`): convert the value to the given unit before displaying; eg `${temperature:F}` displays the temperature in Farhenheit; note that the unit is **not** appended